src/core/net/ip6.cpp

```diff
@@ -1068,6 +1069,22 @@ Error Ip6::SendRaw(OwnedPtr<Message> aMessagePtr)
 
     if (header.GetDestination().IsMulticast())
     {
+#if OPENTHREAD_CONFIG_FILTER_LOCAL_ORIGINATED_MULTICAST_PACKETS
+        // For multicast packets originated from this device, only forward to thread
+        // network if destination is in the multicast listener table
+        if (!aMessagePtr->IsOriginThreadNetif() && Get<ThreadNetif>().HasUnicastAddress(header.GetSource()))
+        {
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_MULTICAST_ROUTING_ENABLE && OPENTHREAD_FTD && \
+    OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+            VerifyOrExit((header.GetDestination().IsMulticastLargerThanRealmLocal() &&
+                          Get<BackboneRouter::MulticastListenersTable>().HasListener(header.GetDestination())),
+                         error = kErrorDrop);
```

Comment:
This seems dropping all multicast packets within realm local scope. Is this desired behavior? My understanding is that the filter should only apply to packets destined to multicast scope greater than realm local.

Reply:
I think when untrusted source is sending to multicast scope <= realm-local, then we should filter it so misbehaving applications can't flood the thread network. The application can still communicate with Thread network with subscribed multicast addresses, similar to the other devices on the backbone interface, their packet sent to multicast address <= realm-local scope are not forwarded to Thread network.

Reply:
I'm inclined not to dropping such packets. If such packets are dropped, that means we cannot run ping command targeting `ff02::1` on Thread interface on a Linux based BR. That's kind of confusing to me.